### DISCUSSION TITLE
Can't load Robot System

### DISCUSSION THREAD

**Discussion by vsardenberg** [visose/Robots#74] [Q&A] [answered]
Hello!

I'm trying to use ROBOTS with our UR-5 in Leibniz University in Hannover.
When I drag and drop Load Robot and plug a value list on it the component becomes red and I receive this error on the balloon:

1. Solution exception:'>' is an unexpected token. The expected token is '='. Line 45, position 94.

Even if I open simple example.gh, the same thing happens. I've copied all libraries to C:\Users\Victor Sardenberg\Documents\Robots\

Any ideas?

Thanks,

Victor Sardenberg

**Comment by visose** [comment]
Hi Victor, it might have to do with the xml you're using for the definition of the UR5. Could you send it to me? (both the xml and the 3dm that you have in the \Robots\ subfolder)

There's a pull request by Ryan Luke Johns that contains a UR5 (https://github.com/visose/Robots/pull/2) but it has to be updated to work on the latest version of the plugin. I can try to add it when I get some free time.

**Comment by vsardenberg** [comment]
Hello visose,

Thanks for your reply. I downloaded all .xml files from this github and I didn't change anything. 
Here you have a link to a copy of my folder:

https://www.dropbox.com/sh/hhd15wy8xr5oev5/AAD_0vbcv8_XCUnDsTeUCz27a?dl=0

Thanks

**Comment by visose** [comment]
Ah, so not even the default libraries are working? Did you follow the instructions in the wiki to install the plugin? Make sure you follow all the points, including the running the Rhino command "GrasshopperDevelopperSettings" and the one about unblocking the .gha and .dll files.

**Comment by vsardenberg** [comment]
Hello Vicente,

I followed every step and double-checked it.

Victor Sardenberg
www.victorsardenberg.com

**Comment by visose** [comment, marked as answer]
That error seems to be from a badly formatted xml file. It's possible you didn't download them correctly (you downloaded GitHub's html file rather than the xml itself. Go to the library,  click on the xml file and then click on the "raw" button (or right click on that button, save as).

I uploaded a library for the University of Leuven that contains a UR5.

**Reply by Habloop** [reply]
![image001](https://user-images.githubusercontent.com/117923003/201118877-60adf4a0-3e9c-47e0-8b0f-466d5f003c0f.png)

How can i solve this? The online libraries do not appear, giving me an error. I tried to manually download a local file as it tells me here: Robot libraries · visose / Robots Wiki · GitHub but nothing. Robot also I downloaded it with the old ways without using the package manager (since the latter does not open it to me giving me an error, I have tried in every way but nothing).

**Reply by visose** [reply]
Hi, could you translate the error message?
Also, we can only help you if you detail the steps you followed and the issues encountered, rather than just saying "nothing".

**Reply by Habloop** [reply]
You're right, forgive me. The image error says: "an error occurred while submitting your request"

So I first installed this plugin not from the package manager but directly by putting all the files into: C: \ Users \ NameUser \ AppData \ Roaming \ Grasshopper \ Libraries \ Robots-1.6.1-rh7_0-any and inside there are gha, dll, yml and files are all unlocked.

(as I said I did not do it from the package manager because the error that gives me when I try to open it tells me "Failed to fetch packages from source (s):" https://yak.rhino3d.com "
Unable to connect to the remote server
Are you sure there's a package server / folder here?

so I install the plugins locally and the Robots file in yak I made it a zip in order to extract the gha dll ecc

So then I opened GH and the plugin was successfully installed. Now my problem arises when I insert the component "load robot system" I right click on "libraries" and then I get this error that I sent you in the image. I also tried to install a local library by following this link but it keeps giving me the same error: https://github.com/visose/Robots/wiki/Robot-libraries. I downloaded a random robot "stuttgart", embedded in both the xml and 3dm in C: \ Users \ userName \ Documents \ after creating the "Robots" folder and inserted the xml and 3dm file, than restarted Rhino but the error continues to exist.

thanks anyway, really

**Reply by visose** [reply]
Do not install the plugin that way, only installing through the package manager is supported now.
It looks like you have an issue with your internet, since you can't access either the yak packages or the robot libraries, which both are unrelated and on different domains.
This could be the firewall in your computer, your university internet blocking too many things, etc. Check if your firewall is blocking the Rhino executable from accessing the internet.
I would fix this instead of trying to work around it, since it will cause additional issues.

**Reply by Habloop** [reply]
Thanks so much for the hint, I solved it. Basically the problem was like you said, the default firewall was not giving access to Rhino. A thousand thanks!

**Comment by vsardenberg** [comment]
Thank you very much. Now everything works nicely.

Victor Sardenberg
www.victorsardenberg.com
